timesketch/lib/analyzers/safebrowsing.py

```diff
@@ -0,0 +1,250 @@
+"""Sketch analyzer plugin for the Safe Browsing API."""
+from __future__ import unicode_literals
+
+import fnmatch
+import logging
+import re
+
+import pkg_resources
+import requests
+from flask import current_app
+
+from timesketch.lib.analyzers import interface, manager
+
+
+class SafeBrowsingSketchPlugin(interface.BaseSketchAnalyzer):
+    """Sketch analyzer for Safe Browsing."""
+
+    NAME = 'safebrowsing'
+
+    _SAFE_BROWSING_BULK_LIMIT = 500
+    _URL_WHITELIST_CONFIG = 'safebrowsing_whitelist.yaml'
+    _SAFEBROWSING_ENTRY_KEEP = frozenset(
+        [
+            'platformType',
+            'threatType',
+        ],
+    )
+
+    _URL_BEGINNING_RE = re.compile(r'(http(s|):\/\/\S*)')
+
+    def __init__(self, index_name, sketch_id):
+        """Initialize The Sketch Analyzer.
+
+        Args:
+            index_name: Elasticsearch index name
+            sketch_id: Sketch ID
+        """
+        super(SafeBrowsingSketchPlugin, self).__init__(index_name, sketch_id)
+
+        self._safebrowsing_api_key = current_app.config.get(
+            'SAFEBROWSING_API_KEY')
+
+        self._google_client_id = current_app.config.get(
+            'SAFEBROWSING_CLIENT_ID',
+            'Timesketch',
+        )
+
+        self._google_client_version = current_app.config.get(
+            'SAFEBROWSING_CLIENT_VERSION',
+            pkg_resources.get_distribution('timesketch').version,
+        )
+
+    def _is_url_whitelisted(self, url, whitelist):
+        """Does a glob-match against the whitelist.
```

Comment:
Indeed, that comment is wrong. Otherwise, we think `fnmatch` is good enough for URL wildcards here. Do you think we should keep the functionality as it is now or should we also replace `fnmatch` with something else?

Reply:
I think fnmatch should be fine here, just to match the docstring to what is actually being done